### ISSUE TITLE
Update ESMF lib in ufs-weather-model for HAFS application

### ISSUE THREAD

**Issue by junwang-noaa** [ufs-community/ufs-weather-model#1107]
## Description
The development work for HAFS moving nest outputs requires a newer version of ESMF library. Once the development finishes, there will be a new ESMF library available and it needs to be committed to ufs-weather-model before the HAFS moving nest output commit.

## Solution
Install the new ESMF 8.3 (version number not decided yet) lib on tier-1 platforms when it's available. Test it and use it in ufs-weather-model
  
## Related to
Directly reference any issues or PRs in this or other repositories that this is related to, and describe how they are related. Example:
- required to support noaa-emc/nems/issues/<issue_number>
- needed by noaa-emc/fv3atm/pull/<pr_number>

**Comment by junwang-noaa** [comment]
ESMF 8.3.0bs09 needs to be installed on ufs-weather-model supported platforms. The library is not expected to change results.

**Comment by junwang-noaa** [comment]
Create an issue hpc-stack.

**Comment by BinLiu-NOAA** [comment]
@junwang-noaa, @DusanJovic-NOAA, and @theurich, Just a quick follow-up. I conducted the ufs-weather-model standard regression tests (rt.conf) with the latest develop branch (as of 03/16/2022) by using the [esmf/v8.3.0b09](https://github.com/esmf-org/esmf/releases/tag/v8.3.0b09) version on Hera. And I got all the regression tests passed successfully. So, unless @theurich tell us to use a newer version of ESMF, we can proceed to install this esmf/v8.3.0b09 tag version under hpc-stack and then update ufs-weather-model to use this version. Thanks!
